contrib/platform/test/com/sun/jna/platform/win32/Kernel32Test.java

```diff
@@ -1020,44 +1018,41 @@ public void testGetFileInformationByHandleEx() throws IOException {
+                    Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileBasicInfo, p, new DWORD(p.size())));
             FILE_BASIC_INFO fbi = new FILE_BASIC_INFO(p);
             // New file has non-zero creation time
             assertTrue(0 != fbi.CreationTime.getValue());
 
             p = new Memory(FILE_STANDARD_INFO.sizeOf());
-            if (false == Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileStandardInfo, p, new DWORD(p.size()))) {
-                fail("GetFileInformationByHandleEx failed with " + Kernel32.INSTANCE.GetLastError());
-            }
+            AbstractWin32TestSupport.assertCallSucceeded("GetFileInformationByHandleEx for FileStandardInfo (" + p.size() + " bytes)",
+                    Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileStandardInfo, p, new DWORD(p.size())));
             FILE_STANDARD_INFO fsi = new FILE_STANDARD_INFO(p);
             // New file has 1 link
             assertEquals(1, fsi.NumberOfLinks);
 
             p = new Memory(FILE_COMPRESSION_INFO.sizeOf());
-            if (false == Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileCompressionInfo, p, new DWORD(p.size()))) {
-                fail("GetFileInformationByHandleEx failed with " + Kernel32.INSTANCE.GetLastError());
-            }
+            AbstractWin32TestSupport.assertCallSucceeded("GetFileInformationByHandleEx for FileCompressionInfo (" + p.size() + " bytes)",
+                    Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileCompressionInfo, p, new DWORD(p.size())));
             FILE_COMPRESSION_INFO fci = new FILE_COMPRESSION_INFO(p);
             // Uncompressed file should be zero
             assertEquals(0, fci.CompressionFormat);
 
             p = new Memory(FILE_ATTRIBUTE_TAG_INFO.sizeOf());
-            if (false == Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileAttributeTagInfo, p, new DWORD(p.size()))) {
-                fail("GetFileInformationByHandleEx failed with " + Kernel32.INSTANCE.GetLastError());
-            }
+            AbstractWin32TestSupport.assertCallSucceeded("GetFileInformationByHandleEx for FileAttributeTagInfo (" + p.size() + " bytes)",
+                    Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileAttributeTagInfo, p, new DWORD(p.size())));
             FILE_ATTRIBUTE_TAG_INFO fati = new FILE_ATTRIBUTE_TAG_INFO(p);
             // New files have the archive bit
             assertEquals(WinNT.FILE_ATTRIBUTE_ARCHIVE, fati.FileAttributes);
 
-            p = new Memory(FILE_ID_INFO.sizeOf());
-            if (false == Kernel32.INSTANCE.GetFileInformationByHandleEx(hFile, WinBase.FileIdInfo, p, new DWORD(p.size()))) {
-                fail("GetFileInformationByHandleEx failed with " + Kernel32.INSTANCE.GetLastError());
+            if (VersionHelpers.IsWindows8OrGreater()) {
```

Comment:
Looks like `FILE_ID_INFO` is also server-only, so you should probably also check `IsWindowsServer()` here.